Features/HideProtectedPosts/functions.php

```diff
@@ -0,0 +1,9 @@
+<?php
+
+namespace Flynt\Features\HideProtectedPosts;
+
+add_action('pre_get_posts', function ($query) {
+    if (!$query->is_singular() && !is_admin()) {
```

Comment:
But `is_singular` takes care of that already
https://codex.wordpress.org/Function_Reference/is_singular